### PR TITLE
Implement a geckolib target

### PR DESCRIPTION
**Pull request by bholley**
[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9209)

### CODE REVIEW

**Comment by bholley** [comment]
Looks like I need to remove the style system's dependency on msg in order to avoid pulling in azure etc. I can do that before or after we land this, as you like.

File: python/tidy.py [hunk 1]
```diff
@@ -38,6 +38,7 @@
     os.path.join(".", "target", "*"),
     os.path.join(".", "ports", "gonk", "src", "native_window_glue.cpp"),
     os.path.join(".", "ports", "cef", "*"),
+    os.path.join(".", "ports", "geckolib", "*"),
```
Comment (SimonSapin) [inline python/tidy.py]: Why exclude geckolib here?

Comment (SimonSapin) [inline python/tidy.py]: r=me with this line remove. I’ve tested locally that `./mach test-tidy` reports no error on this PR’s branch without it.

**Comment by bholley** [comment]
Fixed. I was just copying all the things we did for cef without thinking too much about it.

I may need to ignore some tidy for the bindings (specifically camel-cased stuff), but we can cross that bridge when we come to it.

**Comment by frewsxcv** [comment]
FYI https://github.com/servo/servo/issues/7625

**Comment by KiChjang** [comment]
@bors-servo retry
* Inexplicable timeout on build machine

**Pull request merged**